vyxal/elements.py

```diff
@@ -1011,7 +1011,30 @@ def cosine(lhs, ctx):
 def count_item(lhs, rhs, ctx):
     """Element O
     (any, any) -> returns the number of occurances of b in a
+    (any, fun) -> all elements in a where the result of b(x) is highest
     """
+    if vy_type(lhs) == types.FunctionType:
+        lhs, rhs = rhs, lhs
+    if vy_type(rhs) == types.FunctionType:
+        lhs = iterable(lhs, ctx=ctx)
+        if not lhs:
+            return lhs
+        m = None
+        fun_vals = []
+        for item in lhs:
+            fn_val = safe_apply(rhs, item, ctx=ctx)
+            fun_vals.append(fn_val)
+            if m is None:
+                m = fn_val
+            else:
+                m = dyadic_maximum(m, fn_val, ctx=ctx)
+
+        def gen():
+            for item in lhs:
+                if fun_vals.pop(0) == m:
+                    yield item
+
+        return LazyList(gen())
```

Comment:
no point only wrapping this part - whole part needs wrapping

Reply:
We can return a plain list since we don't need to deal with infinite lists